From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AngeloGioacchino Del Regno <angelogioacchino.delregno@somainline.org>
Date: Thu, 21 Jan 2021 20:05:15 +0100
Subject: arm64: dts: msm8998: Wire up interconnects to MDP and GPU

Wire up the interconnects to both the MDP and the Adreno GPU in
order to get the right balance between performance and power
consumption of both devices.
---
 arch/arm64/boot/dts/qcom/msm8998.dtsi | 21 +++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/msm8998.dtsi b/arch/arm64/boot/dts/qcom/msm8998.dtsi
index 3389957032dd..fe9cf081c989 100644
--- a/arch/arm64/boot/dts/qcom/msm8998.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8998.dtsi
@@ -1438,6 +1438,8 @@ adreno_gpu: gpu@5000000 {
 				"rbcpr",
 				"core";
 
+			interconnects = <&bimc MASTER_OXILI &bimc SLAVE_EBI>;
+			interconnect-names = "gfx-mem";
 			interrupts = <0 300 IRQ_TYPE_LEVEL_HIGH>;
 			iommus = <&adreno_smmu 0>;
 			operating-points-v2 = <&gpu_opp_table>;
@@ -1449,42 +1451,55 @@ gpu_opp_table: opp-table {
 				opp-710000097 {
 					opp-hz = /bits/ 64 <710000097>;
 					opp-level = <RPM_SMD_LEVEL_TURBO>;
+					opp-peak-kBps = <14432000>;
+					opp-avg-kBps = <14400000>;
 					opp-supported-hw = <0xFF>;
 				};
 
 				opp-670000048 {
 					opp-hz = /bits/ 64 <670000048>;
 					opp-level = <RPM_SMD_LEVEL_NOM_PLUS>;
+					opp-peak-kBps = <14432000>;
+					opp-avg-kBps = <14390000>;
 					opp-supported-hw = <0xFF>;
 				};
-
 				opp-596000097 {
 					opp-hz = /bits/ 64 <596000097>;
 					opp-level = <RPM_SMD_LEVEL_NOM>;
+					opp-peak-kBps = <14432000>;
+					opp-avg-kBps = <12440000>;
 					opp-supported-hw = <0xFF>;
 				};
 
 				opp-515000097 {
 					opp-hz = /bits/ 64 <515000097>;
 					opp-level = <RPM_SMD_LEVEL_SVS_PLUS>;
+					opp-peak-kBps = <14432000>;
+					opp-avg-kBps = <10368000>;
 					opp-supported-hw = <0xFF>;
 				};
 
 				opp-414000000 {
 					opp-hz = /bits/ 64 <414000000>;
 					opp-level = <RPM_SMD_LEVEL_SVS>;
+					opp-peak-kBps = <12440000>;
+					opp-avg-kBps = <8136000>;
 					opp-supported-hw = <0xFF>;
 				};
 
 				opp-342000000 {
 					opp-hz = /bits/ 64 <342000000>;
 					opp-level = <RPM_SMD_LEVEL_LOW_SVS>;
+					opp-peak-kBps = <8136000>;
+					opp-avg-kBps = <6144000>;
 					opp-supported-hw = <0xFF>;
 				};
 
 				opp-257000000 {
 					opp-hz = /bits/ 64 <257000000>;
 					opp-level = <RPM_SMD_LEVEL_MIN_SVS>;
+					opp-peak-kBps = <6144000>;
+					opp-avg-kBps = <3296000>;
 					opp-supported-hw = <0xFF>;
 				};
 			};
@@ -2519,6 +2534,10 @@ mdss: mdss@c900000 {
 
 			assigned-clocks = <&mmcc MDSS_MDP_CLK>;
 			assigned-clock-rates = <300000000>;
+			interconnects = <&mnoc MASTER_MDP_P0 &bimc SLAVE_EBI>,
+					<&mnoc MASTER_MDP_P1 &bimc SLAVE_EBI>;
+			interconnect-names = "mdp0-mem",
+					     "mdp1-mem";
 			interrupts = <GIC_SPI 83 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-controller;
 			iommus = <&mmss_smmu 0>;
-- 
Armbian

